securedrop/tests/test_source.py

```diff
@@ -312,6 +312,38 @@ def test_submit_both(self):
             self.assertIn("Thanks! We received your message and document",
                           resp.data)
 
+    @patch('source_app.main.async_genkey')
+    @patch('source_app.main.get_entropy_estimate')
+    def test_submit_message_with_low_entropy(self, get_entropy_estimate,
+                                             async_genkey):
+        get_entropy_estimate.return_value = 300
+
+        with self.client as client:
+            new_codename(client, session)
+            self._dummy_submission(client)
+            resp = client.post('/submit', data=dict(
+                msg="This is a test.",
+                fh=(StringIO(''), ''),
+            ), follow_redirects=True)
+            self.assertEqual(resp.status_code, 200)
+            self.assertFalse(async_genkey.called)
```

Comment:
This last assert is the interesting difference.

Reply:
Oh.... right.